### ISSUE TITLE
Cannot use geometry column or spatial index with space in name against SQL Server

### ISSUE THREAD

**Issue by wicksell** [MapServer/MapServer#6773]
Hi,

When using the plugin for MSSQL it is not possible to have spaces in the geometry column or spatial index name. 

The geometry column name is only escaped in the SELECT list but not in the WHERE clause as shown in the example below.

When parsing the index in mapmssql2008.c/msMSSQL2008LayerParseData, it first search for " using index " and then the next space after that. 

Example:

CONNECTIONTYPE PLUGIN
PLUGIN "msplugin_mssql2008.dll"
DATA "Geometry With Space(geography) from [dbo].[Table With Space] using index [SIndex_Table With Space_Geometry With Space] USING UNIQUE Id USING SRID=4326"

Will result in the following error:

msMSSQL2008LayerGetShape(): Query error. Error executing MSSQL2008 SQL statement: SELECT [Geometry With Space] 'geom',convert(varchar(36), [Id]) 'id' FROM [dbo].[Table With Space] WITH (INDEX([SIndex_Table)) WHERE Geometry With Space.STIntersects(Geography::STGeomFromText('CURVEPOLYGON((18.6374665794468 64.6001461170919,18.6395834608363 64.6001461170919,18.6417003422258 64.6001461170919,18.6417003422258 64.6010552124408,18.6417003422258 64.6019643077897,18.6395834608363 64.6019643077897,18.6374665794468 64.6019643077897,18.6374665794468 64.6010552124408,18.6374665794468 64.6001461170919))',4326)) = 1 
-[Microsoft][ODBC SQL Server Driver][SQL Server]The identifier that starts with 'SIndex_Table)) WHERE Geometry With Space.STIntersects(Geography::STGeomFromText('CURVEPOLYGON((18.6374665794468 64.6001461170919,1' is too long. Maximum length is 128.

Regards
Krister Wicksell

**Comment by jratike80** [comment]
Using "geometry with space" in any database is a call for trouble and an unnecessary risk that I would not take. But you are right that Mapserver should use delimited qualifiers in all places.
Could you test if Mapserver fails now in a similar way if the table name is a reserved keyword https://learn.microsoft.com/en-us/sql/t-sql/language-elements/reserved-keywords-transact-sql?view=sql-server-ver16 and requires  delimiters because of that?

**Comment by wicksell** [comment]
I agree that one should not use spaces in names but if it is possible there will always be people that does.

If defining the table name with delimiters in the data string as in my example above, [dbo].[Table With Space], this is not an issue.

The problem with the geometry column name is that MapServer escape it in some places but not in all so if i escape it in the data string it will be double escaped in the select list.

The issue with the spatial index is just the parser.

**Comment by jratike80** [comment]
Is it possible to use delimiters with the geometry as well, or will it become double-escaped?
DATA "\[Geometry With Space\](geography) from [dbo].[Table With Space] using index [SIndex_Table With Space_Geometry With Space] USING UNIQUE Id USING SRID=4326"`?

**Comment by sdlime** [comment]
Looping in @szekerest to see what he thinks... --Steve

**Comment by wicksell** [comment]
@jratike80 If using delimiters on the geometry column they will be doubled in the select list, SELECT [[Geometry With Space]] 'geom', but OK in the WHERE clause WHERE [Geometry With Space].STIntersects(...

**Comment by jratike80** [comment]
That's what I feared. Maybe we should not try to be too kind for the users by trying to make the DATA line valid by adding delimiters at all. Document that users must use delimited identifiers just like they must use them with their other SQL tools and show a good error message instead.

**Comment by szekerest** [comment]
I think we should just check if the prop name is already escaped in msMSSQL2008LayerEscapePropertyName.
